test/unit/test_annotated_code.c

```diff
@@ -0,0 +1,305 @@
+	strcpy (test_string, "\nvoid main(void)\n{\n    sym.imp.puts(\"Hello, World!\");\n    return;\n}\n");
+	RAnnotatedCode *code = r_annotated_code_new (test_string);
+
+	RVector /*<RCodeAnnotation>*/ *test_annotations;
+	test_annotations = get_annotations_for_hello_world ();
+	RCodeAnnotation *annotation;
+	r_vector_foreach (test_annotations, annotation) {
+		r_annotated_code_add_annotation (code, annotation);
+	}
+	return code;
+}
+
+static bool test_r_annotated_code_new() {
+	//Testing RAnnoatedCode->code
+	char *test_string = malloc (14 * sizeof (char));
+	strcpy (test_string, "How are you?");
+	RAnnotatedCode *code = r_annotated_code_new (test_string);
+	mu_assert_streq (code->code, test_string, "Code in RAnnotatedCode is not set as expected");
+
+	// Testing RAnnoatedCode->annotations
+	mu_assert_eq (code->annotations.elem_size, sizeof (RCodeAnnotation), "Code Annotations are initialized is not properly");
+
+	mu_end;
+}
+
+static bool test_r_annotated_code_free() {
+	char *test_string = malloc (14 * sizeof (char));
+	strcpy (test_string, "How are you?");
+	RAnnotatedCode *code = r_annotated_code_new (test_string);
+
+	RCodeAnnotation test_annotation1, test_annotation2;
+	test_annotation1 = make_code_annotation (1, 2, R_CODE_ANNOTATION_TYPE_OFFSET, 123, R_SYNTAX_HIGHLIGHT_TYPE_KEYWORD);
+	r_vector_push (&code->annotations, &test_annotation1);
+	test_annotation2 = make_code_annotation (1, 5, R_CODE_ANNOTATION_TYPE_SYNTAX_HIGHLIGHT, 123, R_SYNTAX_HIGHLIGHT_TYPE_KEYWORD);
+	r_vector_push (&code->annotations, &test_annotation2);
+
+	// This test is only for run errors
+	r_annotated_code_free (code);
+
+	mu_end;
+}
+
+static bool test_equal(RCodeAnnotation *first, RCodeAnnotation *second) { // First - Got, Second - Expected
+	mu_assert_eq (first->start, second->start, "start of annotations doesn't match");
+	mu_assert_eq (first->end, second->end, "end of annotations doesn't match");
+	mu_assert_eq (first->type, second->type, "type of annotation doesn't match");
+	if (first->type == R_CODE_ANNOTATION_TYPE_OFFSET) {
+		mu_assert_eq (first->offset.offset, second->offset.offset, "offset of annotations doesn't match");
+	}
+	if (first->type == R_CODE_ANNOTATION_TYPE_SYNTAX_HIGHLIGHT) {
+		mu_assert_eq (first->syntax_highlight.type, second->syntax_highlight.type, "syntax highlight type of offset doesn't match");
+	}
+}
+
+static bool test_r_annotated_code_add_annotation() {
+	char *test_string = malloc (54 * sizeof (char));
+	strcpy (test_string, "abcdefghijklmnopqrtstuvwxyz ABCDEFGHIJKLMNOPQRSTUVWXYZ");
+	RAnnotatedCode *code = r_annotated_code_new (test_string);
+	RVector /*<RCodeAnnotation>*/ *test_annotations;
+	test_annotations = get_some_code_annotation_for_add ();
+	RCodeAnnotation *annotation;
+	r_vector_foreach (test_annotations, annotation) {
+		r_annotated_code_add_annotation (code, annotation);
+	}
+
+	//Comparing
+	test_equal (r_vector_index_ptr (&code->annotations, 0), r_vector_index_ptr (test_annotations, 0));
```

Comment:
We want to break the test if this fails:
```suggestion
	if (!test_equal (r_vector_index_ptr (&code->annotations, 0), r_vector_index_ptr (test_annotations, 0))) {
		return false;
	}
```
Same goes for all the other usages of `test_equal()`.

Reply:
Changed as suggested.